ClientFactory/ClientFactoryInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Http\HttplugBundle\ClientFactory;
+
+use Http\Client\HttpClient;
+
+/**
+ * Interface ClientFactoryInterface.
+ */
+interface ClientFactoryInterface
+{
+    /**
+     * Configure and return a client.
+     *
+     * @param array $config
+     *
+     * @return HttpClient
```

Comment:
what should happen if this does not work out? i would rather throw an exception than allow returning null